python/lsst/ip/diffim/imageDecorrelation.py

```diff
@@ -0,0 +1,191 @@
+
+    def post_conv_psf(psf, kernel, sig1=1., sig2=1.):
+        kft = post_conv_psf_ft2(psf, kernel, sig1, sig2)
+        out = ifft2(kft)
+        return out
+
+    im2_psf_small = im2_psf
+    # First compute the science image's (im2's) psf, subset on -16:15 coords
+    if im2_psf.shape[0] > 50:
+        x0im, y0im = getImageGrid(im2_psf)
+        x = np.arange(-16, 16, 1)
+        y = x.copy()
+        x0, y0 = np.meshgrid(x, y)
+        im2_psf_small = im2_psf[(x0im.max()+x.min()+1):(x0im.max()-x.min()+1),
+                                (y0im.max()+y.min()+1):(y0im.max()-y.min()+1)]
+    pcf = post_conv_psf(psf=im2_psf_small, kernel=kappa, sig1=sig2, sig2=sig1)
+    pcf = pcf.real / pcf.real.sum()
+    return pcf
+
+
+def performExposureDecorrelation(templateExposure, exposure, subtractedExposure, psfMatchingKernel, log):
+    """! Compute decorrelation correction on an image difference exposure, given the
+    input template/science exposures and the (potentially spatially-varying) PSF matching kernel.
+
+    @param templateExposure the template afwImage.Exposure used for PSF matching
+    @param exposure the science afwImage.Exposure used for PSF matching
+    @param subtractedExposure the resulting subtracted exposure produced by 
+    ip_diffim.ImagePsfMatchTask.subtractExposures()
+    @param psfMatchingKernel an (optionally spatially-varying) PSF matching kernel produced 
+    by ip_diffim.ImagePsfMatchTask.subtractExposures()
+    @param log an input pexLog for outputting messages
+
+    @return the corrected subtractedExposure (updated in-place) with an updated PSF, and
+    the decorrelation correction kernel (which may be ignored)
+
+    @note Here we currently convert a spatially-varying matching kernel into a constant kernel, 
+    just compute it for the center of the image.
+    @note Still TBD: do we need to modify the new subtractedExposure's variance plane?
+    """
+    kimg = None
+    try:
+        if psfMatchingKernel.isSpatiallyVarying():
+            spatialKernel = psfMatchingKernel
+            kimg = afwImage.ImageD(spatialKernel.getDimensions())
+            bbox = subtractedExposure.getBBox()
+            xcen = (bbox.getBeginX() + bbox.getEndX()) / 2.
+            ycen = (bbox.getBeginY() + bbox.getEndY()) / 2.
+            spatialKernel.computeImage(kimg, True, xcen, ycen)
+    except:  # Not a spatially-varying kernel (usually not true)
+        kimg = psfMatchingKernel.computeImage()
+
+    # Compute the images' sigmas (sqrt of variance)
+    sig1 = templateExposure.getMaskedImage().getVariance().getArray()
+    sig1squared, _ = computeClippedImageStats(sig1)
+    sig1 = np.sqrt(sig1squared)
+
+    sig2 = exposure.getMaskedImage().getVariance().getArray()
+    sig2squared, _ = computeClippedImageStats(sig2)
+    sig2 = np.sqrt(sig2squared)
+
+    corrKernel = computeDecorrelationCorrectionKernel(kimg.getArray(), sig1=sig1, sig2=sig2)
+    # Eventually, use afwMath.convolve(), but for now we just use scipy.
```

Comment:
This worries me. The afw convolution code is supposed to be easy to use, and has the advantage of dealing with the variance and mask planes correctly. Why not use it?